rg_instructor_analytics/tests/test_view.py

```diff
@@ -0,0 +1,149 @@
+    @patch('rg_instructor_analytics.views.EnrollmentStatisticView.get_statistic_per_day')
+    def test_enroll_statistic_post_call(
+            self, moc_get_statistic_per_day,
+            moc_course_key_from_string,
+            moc_get_course_by_id,
+            moc_has_access):
+        """Verify standard post flow."""
+        moc_get_statistic_per_day.return_value = self.EXPECTED_ENROLL_STATISTIC
+        moc_course_key_from_string.return_value = 'key'
+        moc_get_course_by_id.return_value = 'course'
+        moc_has_access.return_value = True
+
+        request = self.factory.post('/courses/' + self.MOCK_COURSE_ID + '/tab/instructor_analytics/api/enroll_statics/')
+        request.user = self.MOCK_ALLOWED_USER
+        request.POST = self.MOCK_REQUEST_PARAMS
+
+        response = EnrollmentStatisticView.as_view()(request, course_id=self.MOCK_COURSE_ID)
+
+        self.assertEqual(response.status_code, 200)
+        moc_course_key_from_string.assert_called_once_with(self.MOCK_COURSE_ID)
+        moc_has_access.assert_called_once_with(self.MOCK_ALLOWED_USER, 'staff', moc_get_course_by_id.return_value)
+
+    @patch('rg_instructor_analytics.views.log.error')
+    @patch('rg_instructor_analytics.views.CourseKey.from_string')
+    def test_enroll_statistic_post_call_with_fake_course(
+            self,
+            moc_course_key_from_string,
+            moc_log_error,
+    ):
+        """Verify reaction to the invalid course."""
+        moc_course_key_from_string.return_value = 'key'
+        moc_course_key_from_string.side_effect = Mock(side_effect=InvalidKeyError('', ''))
+
+        request = self.factory.post('/courses/' + self.MOCK_COURSE_ID + '/tab/instructor_analytics/api/enroll_statics/')
+        request.user = self.MOCK_ALLOWED_USER
+        request.POST = self.MOCK_REQUEST_PARAMS
+
+        response = EnrollmentStatisticView.as_view()(request, course_id=self.MOCK_COURSE_ID)
+
+        self.assertEqual(response.status_code, 400)
+        moc_log_error.assert_called_once_with(
+            "Unable to find course with course key %s while getting enrollment statistic",
+            self.MOCK_COURSE_ID
+        )
+
+    @patch('rg_instructor_analytics.views.log.error')
+    @patch('rg_instructor_analytics.views.has_access')
+    @patch('rg_instructor_analytics.views.get_course_by_id')
+    @patch('rg_instructor_analytics.views.CourseKey.from_string')
+    def test_enroll_statistic_post_call_with_unallowed_user(
+            self,
+            moc_course_key_from_string,
+            moc_get_course_by_id,
+            moc_has_access,
+            moc_log_error
+    ):
+        """Verify reaction to user, which do not have access to  the given API."""
+        moc_course_key_from_string.return_value = 'key'
+        moc_get_course_by_id.return_value = 'course'
+        moc_has_access.return_value = False
+
+        request = self.factory.post('/courses/' + self.MOCK_COURSE_ID + '/tab/instructor_analytics/api/enroll_statics/')
```

Comment:
I would add some BASE_TEST class with request factory creation to avoid code duplication in almost every test_function, or at least move the same code to the `setUp` method

Reply:
Move initialization of request to the setup method